app/src/main/java/io/pumpkinz/pumpkinreader/NewsListFragment.java

```diff
@@ -126,9 +125,78 @@ public void onError(Throwable e) {
             @Override
             public void onNext(List<News> items) {
                 Log.d("stories", String.valueOf(items.size()));
+
+                NewsAdapter adapter = (NewsAdapter) newsList.getAdapter();
+                int lastItemIdx = adapter.getItemCount() - 1;
+
+                if (lastItemIdx >= 0 && adapter.getItem(lastItemIdx) == null) {
+                    adapter.removeItem(lastItemIdx);
+                }
+
                 newsAdapter.addDataset(items);
             }
         });
     }
 
+    private Observable<List<News>> loadNewsData(int from, int count, boolean refresh) {
+        Observable<List<News>> ret;
+
+        switch (this.newsType) {
+            case R.string.top:
+                ret = dataSource.getHNTop(from, count, refresh);
+                break;
+            case R.string.recent:
+                ret = dataSource.getHNNew(from, count, refresh);
+                break;
+            case R.string.ask_hn:
+                ret = dataSource.getHNAsk(from, count, refresh);
+                break;
+            case R.string.show_hn:
+                ret = dataSource.getHNShow(from, count, refresh);
+                break;
+            case R.string.job:
+                ret = dataSource.getHNJob(from, count, refresh);
+                break;
+            default:
+                ret = dataSource.getHNTop(from, count, refresh);
+        }
+
+        return ret.cache();
+    }
+
+    private void setUpNewsList(View view) {
+        newsList = (RecyclerView) view.findViewById(R.id.news_list);
+
+        final LinearLayoutManager layoutManager = new LinearLayoutManager(getActivity());
+        layoutManager.setOrientation(LinearLayoutManager.VERTICAL);
```

Comment:
It's different than what you think.

Vertical - Horizontal is the flow of child widget in the layout.
Portrait - Landscape is the orientation of the screen.